base_order_by_related/models/order_by_related_mixin.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2023 Therp (<http://www.therp.nl>).
+# @author Tom Blauwendraat <tblauwendraat@therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class OrderByRelatedMixin(models.AbstractModel):
+    _name = "order.by.related.mixin"
+    _description = "Mixin to allow ordering by non-stored related fields"
+
+    @api.model
+    def _related_join_calc(self, alias, fname, query, real_fname):
+        model, field = self, self._fields[fname]
+        while field.related:
+            # retrieve the parent model where field is inherited from
+            related_model = self.env[field.related_field.model_name]
```

Comment:
```suggestion
            if field.related_field.model_name not in self.env:
                continue
            related_model = self.env[field.related_field.model_name]
```

for this kind of low-level stuff I'd suggest to be cautious about models/fields not existing due to a half loaded registry, also above with the field